src/app/chat/view.nim

```diff
@@ -996,3 +998,23 @@ QtObject:
   proc formatInputCode(self: ChatsView, inputText: string): string {.slot.} =
     let strikeThroughRegex = re"""(?<!\>)`(?!<span style=" font-family:'monospace';">)([^*]+)(?!<\/span>)`"""
     self.formatInputStuff(strikeThroughRegex, inputText)
+
+  proc createCommunityChannel*(self: ChatsView, communityId: string, name: string, description: string, private: bool): string {.slot.} =
```

Comment:
Why was this moved from `views/communities` to  `view.nim`?

Reply:
This was moved so that we could call `setActiveChannel` on the created/edited channel in the chats view (`view.nim`). The only way to achieve this from within the communities view (`views/communities.nim`) was to use the event emitter to emit an event from the communities view, but that can get messy and I would prefer to avoid use of the event emitter if possible.